### PR TITLE
Docs: Update the README with motivation and future intentions.

### PR DESCRIPTION
**Pull request by SamirTalwar**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,28 @@
 
 *Say the magic word.*
 
+`plz` is a job runner targetted at training machine learning models as cheaply as possible.
+It is, however, fairly general, so can be used for practically anything that requires running a job.
```
Comment (sergiogiro) [inline README.md]: 'a job in the cloud'? 'a job in a repeatable fashion'? The description should give you an idea of what it does for you (beyond 'running a job', which is not very informative)

File: README.md [hunk 2]
```diff
@@ -2,6 +2,28 @@
...
+
+`plz run` performs a number of operations:
+
+1. It starts a "worker" (typically on AWS EC2, as a spot instance) to run your job.
+2. It packages your code and data and ships them to the worker.
```
Comment (sergiogiro) [inline README.md]: "ships it"?

File: README.md [hunk 3]
```diff
@@ -2,6 +2,28 @@
...
+We have tried to strike a balance between speed and price, so the `plz` controller keeps instances around for a little while before terminating them.
+
+## Future work
+
+In the future, `plz` is intended to:
+
+* gather and plot metrics over multiple runs, to understand how training is progressing,
+* search over hyperparameters in parallel to optimise models,
+* manage epochs to capture intermediate metrics and results, and terminate runs early,
+* and whatever else is requested of us!
```
Comment (sergiogiro) [inline README.md]: I'd add "(within reason)" at the end. If at all for the humorous effect :P

File: README.md (hunk 2 above)
Comment (SamirTalwar) [reply README.md]: "Code and data" is plural.

**Pull request merged**